switchapi/src/switch_packet.c

```diff
@@ -163,6 +163,13 @@ static void switch_packet_rx_from_hw() {
     if (packet_header->cpu_header.reason_code ==
         SWITCH_HOSTIF_REASON_CODE_NULL_DROP)
       continue;
+    if (packet_header->cpu_header.reason_code ==
```

Comment:
I think the memcpy before these lines need to be adjusted to remove rcode specific header from the packet.
We could call len = extract_rcode_header(rcode, packet, &rcode_header) where we can copy a set of bytes from packet and then use len to skip over when copying the packet. We could make len as part of rcode_header (may be better).